treefmt.nix

```diff
@@ -1,14 +1,22 @@
 {
-  perSystem = { pkgs, ... }: {
+  perSystem = { config, pkgs, ... }: {
     treefmt = {
       # Used to find the project root
       projectRootFile = "flake.lock";
 
+      programs.deadnix.enable = true;
```

Comment:
afaik treefmt only supports one formatter per extension. so I don't think enabling this option will have any effect or it will disable existing formatters below depending on the order.

Reply:
technically it works, but both formatters might touch the same file at the same time

Reply:
sounds racy.

Reply:
in practice I never had a problem, but yeah

Reply:
The proper (and configurable) fix would be sequences which has an open issue in treefmt but could a "cheaper" fix be making the order of formatters deterministic and doing something like `a_deadnix` `b_nixpkgs`?

Reply:
That doesn't allow separate excludes for deadnix which is what I was trying for with this approach.

Reply:
I guess adding a ignore directive to deadnix would also work and would allow us to have deadnix, statix and nixpkgs-fmt all under the nix formatter.

Reply:
https://github.com/astro/deadnix/commit/03b7f9a84086c58be91245a589c8c9144ffa6e2e

Upsteam added `#deadnix: skip` so we can have deadnix under `nix` once it lands in nixpkgs.